network/src/transport/mod.rs

```diff
@@ -147,7 +147,7 @@ impl ConnectionMetadata {
             remote_peer_id,
             role,
             origin,
-            connection_id: ConnectionId::default(),
+            connection_id: CONNECTION_ID_GENERATOR.next(),
```

Comment:
I don't know why we need this change 😆 I assume this doesn't affect anything (given that it's test only)?

Reply:
This is a super annoying problem with the `ConnectionId`.

I had the option of adding another constructor, making the object public or crate public, and routing a bunch of stuff through so we can have `ConnectionId` comparison in tests.

Or, just increment it here, so we can have different `ConnectionId`s and not always have the same "default" mock one.